### ISSUE TITLE
Session's reset! is causing a new browser to launch if the browser has already been quit.

### ISSUE THREAD

**Issue by sbonebrake** [jnicklas/capybara#1237]
Currently session's reset! is as follows:
```ruby
def reset!
      if @touched
        driver.reset!
        assert_no_selector :xpath, "/html/body/*"
        @touched = false
      end
      raise @server.error if Capybara.raise_server_errors and @server and @server.error
    ensure
      @server.reset_error! if @server
    end
```
If driver.quit has been called sometime during the test, the call to assert_no_selector will eventually cause the browser to get initialized again.  This is causing an additional Saucelabs session to open during the rspec.rb's config.after method.
It seems to me that the reset! method should be something like:
```ruby
def reset!
      if @touched && driver.browser_initialized?
      ...
```

**Comment by chris1248** [comment]
Can we get some resolution on this? Is there a pull request pending?

**Comment by twalpole** [comment]
This was released in 2.4.2
